DateChooser: even better date/time widgets

cyrilRomain opened the pull request:
Major rewrite except the calendar display algorithm.

New API:
 - InputDate: date chooser
 - InputTime: time chooser
 - InputDateTime: date and time chooser

Features improvments:
 - display sundays on right
 - add a today goto button 'o' (similar to thunderbird)
 - highlight today, selected and hovered days (w/ different colors)

Code:
 - removed datechooser original implementation
 - removed SetDate* functions, it is up to the user to setup struc tm
 - kept the calendar display algorithm
 - support more strict compilation

Examples:
 - imguidatechooser to imgui_datetime

Flix01 commented:
@cyrilRomain, thank you for your effort!

I still have to test it, but I think that having three widgets is better than one!

Some notes:

- Maybe renaming it to imguidatetimechooser.h/.cpp is more consistent with other addons (and also adding "Chooser" to the function names) [*].
- Displaying sundays on right might not be what people want (in many cultures, and from an historical point of view, sunday is the first day of the week [[see the wiki about it](https://en.wikipedia.org/wiki/Sunday)]).
- A screenshot of the new widgets could be handy to see the result.

But, aside from these things, I think it's a useful improvement. Thank you again for having spent your time on it.

**Edit:** [*] Maybe imguiinputdatetime.h/.cpp or just imguidatetime.h/.cpp could be OK. and having the "Input" prefix is more consistent with other Imgui widgets... so it was a good choice not to append "Chooser" when "Input" is used. 
In short: forget point 1: just removing the underscore in imgui_datetime is OK for me.

cyrilRomain commented:
@Flix01 thank for your feedback

- added a commit to address consistency point
- regarding sundays I followed [ISO 8601](https://en.wikipedia.org/wiki/ISO_week_date). I added a commit though to enable sunday first, but w/o customization i.e. user has to change the hardcoded sunday_first variable.
- [DateTime screenshot](https://drop.greenmycity.eu/f.php?h=1z9G3j6v&p=1) (available for download for 1 month)

@Flix01's edit: (pasted the screenshot here):
![image](https://user-images.githubusercontent.com/9608982/178286655-6030d872-db59-4d91-9c73-fde81f443d7d.png)

Flix01 commented:
Seems OK to me. 
Just one more thing before merging it:

Can you please modify the code this way?
```c++
#ifdef IMGUIDATETIME_DISPLAYSUNDAYFIRST
static bool sunday_first = true; // display sundays first (like before)
#else
static bool sunday_first = false; // display mondays first as ISO 8601 suggest
#endif
```

cyrilRomain commented:
Good point.
Fix done in last commit (with file renaming I forgot to rename back w/o "chooser").
Feel free to squash when merging.
